clrenv/lazy_env.py

```diff
@@ -184,6 +222,7 @@ def _apply_functions(d, recursive=False):
       ^parameter: Looks up the given value in AWS Parameter store.
     """
     new = Munch()
+    global _parameter_store_key_cache
```

Comment:
looks like we're overusing globals here. Why can't this just be a local variable that we pass into `resolve_parameters`?

Reply:
This function is recursive, and we want to use the same common cache. The alternative would be to change this function into an iterative tree search algorithm.